.expeditor/end_to_end.pipeline.yml

```diff
@@ -112,3 +112,16 @@ steps:
           environment:
             - BUILD_PKG_TARGET=x86_64-linux
             - HAB_BLDR_URL=$ACCEPTANCE_HAB_BLDR_URL
+
+  - label: "[:linux: test_studio_auto_installs]"
+    command:
+      - .expeditor/scripts/setup_environment.sh DEV
+      - hab origin key generate $HAB_ORIGIN
```

Comment:
Why isn't this in the e2e script?

Reply:
Originally it was, but I found that iterating on it cluttered up my local workstation with fake keys.  Since we don't have a great way to check for a key or conditionally generate one at the moment, I decided to move it into the pipeline steps.  

Locally, you're almost always guaranteed to have a key generated for your default origin and I'd argue that we should leave it up to a user to do that.  In the pipeline however we can't make that guarantee and a key is required, otherwise the studio creates but exits `1` because it has no keys to import. 

With https://github.com/habitat-sh/habitat/pull/6855 this may not be an issue anymore in which case I'd agree it should go back into the script. I'd need to verify that keys won't make it out of that container. I don't think they will, but haven't looked closely at that PR yet.

Reply:
If the assumption is there will always be a key available, we may want to move this to `setup_environment` - thoughts?

Reply:
I think that assumption is only true in `studio` or `pkg build` cases.  Other cases keys will be fetched as needed from builder. Are we worried about having keys on disk when we don't need them?  The only case I can think of is if we add e2e tests around key generation.   There also may be cases where it would mask behavior like we saw with the studio that lead to this test :smile: , but I can't think of what they might be so are probably aren't real concerns. 

Is the intent that a user runs setup_environment locally when adding/iterating on things? If so, my comments about key generation spam and #6855 remain the same. 

Otherwise, that change seems reasonable to me if we don't care that there is a key generation step when we don't need it.